### ISSUE TITLE
GDI Handle leaks: TabSkins::Initialize

### ISSUE THREAD

**Issue by fire-eggs** [rainmeter/rainmeter#208]
Every invocation of the "Manage Rainmeter" dialog currently leaks four GDI objects. This is due to a failure to release the `ImageList` created in `TabSkins::Initialize`.

`ImageList_Destroy` must be called to release the `ImageList`, presumably in `TabSkins::~TabSkins()`.

Found via observation; diagnosed with DeLeaker. Note: built the current master branch, last modified on Oct 9, 2019.

**Comment by brianferguson** [comment]
While not positive, I believe the flag `LR_SHARED` used in the [LoadImage](https://github.com/rainmeter/rainmeter/blob/master/Library/DialogManage.cpp#L579) function will tell the system to destroy the resource when it is no longer needed.

[https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-loadimagea#lr_shared](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-loadimagea#lr_shared)

I believe the Manage dialog can only be invoked 1 time while Rainmeter is running...meaning I do not believe the actual "window" is destroyed (once created) until Rainmeter has exited. So the resource may appear to have leaked when the Manage dialog was closed, when it was never actually destroyed yet. I believe all of our dialogs operate in this manner. You can test this by opening the About dialog, clicking on the Plugins tab which will populate all the external plugins (this usually takes a few seconds). You can then close the About dialog, re-open it again and click on the Plugins tab (it should load very quickly, since the dialog was never destroyed prior to re-opening).

-Brian

**Comment by fire-eggs** [comment]
Thanks, Brian. My experience is different.

The `TabSkins::Initialize` method is invoked each and every time the "Manage" dialog appears. The `TabSkins` destructor is invoked each and every time the "Manage" dialog is closed. I verified this with the debugger and my local build of RainMeter.

Secondly, I'm not referring to `LoadImage`, I'm referring to `ImageList_Create`. Currently, there is no corresponding call to `ImageList_Destroy`, required as documented. So even if the `TabSkins` code was invoked only once for the lifetime of RainMeter, currently there are four handles leaked. Since that is _not_ the case, as a new RainMeter user, I frequently invoked the "Manage" dialog to experiment with different skins and settings, the handle leak is very noticeable.
Kevin

**Comment by fire-eggs** [comment]
>While not positive, I believe the flag LR_SHARED used in the LoadImage function will tell the system to destroy the resource when it is no longer needed.

Correct. However, by failing to call ImageList_Destroy, the ImageList still exists, still has a reference to those GDI objects, and the resource is "still needed".

**Comment by brianferguson** [comment]
> Secondly, I'm not referring to `LoadImage`, I'm referring to `ImageList_Create`. Currently, there is no corresponding call to `ImageList_Destroy`, required as documented. 

Hmmm, while I realized you were talking about ImageList_Create, I assumed the "objects" you talking about was the icon images handles themselves, which I do believe are being destroyed when no longer referenced as per the link.

I guess whoever wrote this particular piece of code thought the purpose of calling `ImageList_Destroy` was to clean up any image resources within the ImageList itself. Since loading that image(s) with the `LR_SHARED` flag (thus having the system clean it up when no longer needed), I can only assume that it was thought the clean up was being done properly when the resource was no longer used. It makes sense that there are some underlying "objects" created by the ImageList itself that would need to be cleaned up.

I assume we will have to keep track of the image list handle for cleanup in the destructor (sorry, thinking out loud :grin: ). 

I will investigate this further...

-Brian
